src/emcee.jl

```diff
@@ -0,0 +1,223 @@
+struct Ensemble{D} <: MHSampler
+    n_walkers::Int
+    proposal::D
+end
+
+# Define the first step! function, which is called at the 
+# beginning of sampling. Return the initial parameter used
+# to define the sampler.
+function AbstractMCMC.step!(
+    rng::Random.AbstractRNG,
+    model::DensityModel,
+    spl::Ensemble,
+    N::Integer,
+    ::Nothing;
+    init_params = nothing,
+    kwargs...,
+)
+    if init_params === nothing
+        return propose(rng, spl, model)
+    else
+        return Transition(model, init_params)
+    end
+end
+
+# Define the other step functions. Returns a Transition containing
+# either a new proposal (if accepted) or the previous proposal 
+# (if not accepted).
+function AbstractMCMC.step!(
+    rng::Random.AbstractRNG,
+    model::DensityModel,
+    spl::Ensemble,
+    ::Integer,
+    params_prev;
+    kwargs...,
+)
+    # Generate a new proposal. Accept/reject happens at proposal level.
+    return propose(rng, spl, model, params_prev)
+end
+
+#
+# Initial proposal
+# 
+function propose(rng::Random.AbstractRNG, spl::Ensemble, model::DensityModel)
+    # Make the first proposal with a static draw from the prior.
+    static_prop = StaticProposal(spl.proposal.proposal)
+    mh_spl = MetropolisHastings(static_prop)
+    return [propose(rng, mh_spl, model) for _ in 1:spl.n_walkers]
+end
+
+
+
+#
+# Every other proposal
+# 
+function propose(rng::Random.AbstractRNG, spl::Ensemble, model::DensityModel, walkers::Vector{W}) where {W<:Transition}
+    new_walkers = Vector{W}(undef, spl.n_walkers)
+    interval = 1:spl.n_walkers
+
+
+    nwalkers = spl.n_walkers
+    others = 1:(nwalkers - 1)
+    for i in interval
```

Comment:
Maybe one could rather rely on `length(walkers)` since we want to access every element of `walkers` and update all entries of `new_walkers`:
```suggestion
    nwalkers = length(walkers)
    others = 1:(nwalkers - 1)
    for i in 1:nwalkers
```